pandas/tests/io/excel/test_readers.py

```diff
@@ -1044,3 +1044,9 @@ def test_excel_read_binary(self, engine, read_ext):
 
         actual = pd.read_excel(data, engine=engine)
         tm.assert_frame_equal(expected, actual)
+
+    def test_excel_high_surrogate(self, read_ext):
+        # GH 23809
+
+        # should not produce a segmentation violation
+        pd.read_excel("high_surrogate.xlsx")
```

Comment:
Can you create the expected dataframe and use `tm.assert_frame_equal`?

Reply:
Added